### DISCUSSION TITLE
Opening git plugin in the right pane

### DISCUSSION THREAD

**Discussion by rustysys-dev** [eclipse-theia/theia#9402] [Q&A]
When installing Theia using `yarn` the  git plugin seems to be automatically included in the base configuration.  This is great, however, unlike vscode, I would like to have this git panel... and really all other plugins on the right side (I don't want my tree to be hidden when performing these tasks)

What would it take, to do this?  Is this a configuration issue or would something need to be recoded? and if so, what?

If anyone could point me in the right direction I would be grateful!!

**Comment by vince-fugnitto** [comment]
@rustysys-dev thank you for the discussion!

If I understand you correctly, you are referring to the `SCM` view which is by default added to the shell layout, and it's placed on the left sidebar:

The following defined the [widget-options](https://eclipse-theia.github.io/theia/docs/next/interfaces/core.applicationshell-2.widgetoptions.html), namely that it should be added to the `left`:

https://github.com/eclipse-theia/theia/blob/a8dec8d798342631f2ab3a84bec4b8fdc4621d61/packages/scm/src/browser/scm-contribution.ts#L99-L111

The following adds the view to the default layout (if not other layout exists) [initializeLayout](https://eclipse-theia.github.io/theia/docs/next/interfaces/core.frontendapplicationcontribution-1.html#initializelayout):

https://github.com/eclipse-theia/theia/blob/a8dec8d798342631f2ab3a84bec4b8fdc4621d61/packages/scm/src/browser/scm-contribution.ts#L118-L120

If you'd like to add the view to the `right`, you would need to implement a [custom theia extension](https://theia-ide.org/docs/authoring_extensions/) which updates this behavior by rebinding the view to your custom-view which defines new options.

As for all views besides the explorer/navigator being on the right going forward, you might want to update the default behavior which handles `left` and `right` views.

https://github.com/eclipse-theia/theia/blob/a8dec8d798342631f2ab3a84bec4b8fdc4621d61/packages/core/src/browser/shell/application-shell.ts#L742-L760
